src/flamegraph/color/palettes.rs

```diff
@@ -59,6 +59,31 @@ pub(super) mod perl {
     }
 }
 
+pub(super) mod python {
+    use crate::flamegraph::color::BasicPalette;
+
+    fn split_any_path(path: &str) -> impl Iterator<Item = &str> {
+        path.split(|c| c == '/' || c == '\\')
+    }
+
+    pub(in super::super) fn resolve(name: &str) -> BasicPalette {
+      if split_any_path(name).any(|part| part == "site-packages") {
+            return BasicPalette::Mem;
```

Comment:
Usually it's better to refer to color palettes here rather than "semantic" palettes. Makes it easier to ensure the colors chosen for different things don't overlap.

Also, I'd probably skip the `return` here and below, and just stick the last line of the function in an `} else {`.